account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -84,24 +86,161 @@
 			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
 		}
 
-		UserBean userBean = userMapper.userWSToBean(userWS);
+		UserBean userBean = userMapper.toBean(userWS);
 
 		UserBean user = accountService.createUser(userBean);
 
 		try {
-			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.REGISTRATION.name());
+			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.ACTIVATING.name());
 		} catch (MessagingException e) {
 			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.EMAIL_MESSAGING_EXCEPTION.name(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
 		}
+		return new ResponseEntity<>(userMapper.toWS(user), HttpStatus.CREATED);
+	}
+
+	@GetMapping("/activate/{token}")
+	public ResponseEntity<?> activateAccount(@PathVariable("token") String token) {
+		Optional<UserBean> user = accountService.findByToken(token);
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<Object>(new ApiErrorsWS(ErrorCode.INVALID_TOKEN.name(), "Invalid Token."), HttpStatus.BAD_REQUEST);
+		}
+		accountService.activateUser(user.get().getId());
 
-		return new ResponseEntity<>(userMapper.userBeanToWS(user), HttpStatus.CREATED);
+		// TODO: need to tell FE to redirect to /login page after activate signup token
```

Comment:
Remove this comment. How FE flow works should be independent of BE response.

Reply:
fix